src/path_manager.c

```diff
@@ -105,86 +105,158 @@ static char const *get_pm_name(void const *data, size_t len)
 }
 #endif // MPTCPD_ENABLE_PM_NAME
 
-static void handle_connection_created(struct l_genl_msg *msg,
-                                      void *user_data)
+
+/**
+ * @struct pm_event_attrs
+ *
+ * @brief MPTCP generic netlink attribute values.
+ */
+struct pm_event_attrs
+{
+        /// MPTCP connection token.
+        mptcpd_token_t const *token;
+
+        /// Local MPTCP address ID.
+        mptcpd_aid_t const *laddr_id;
+
+        /// Remote MPTCP address ID.
+        mptcpd_aid_t const *raddr_id;
+
+        /// Local IPv4 address.
+        in_addr_t laddr4;
+
+        /// Remote IPv4 address.
+        in_addr_t raddr4;
+
+        /// Local IPv6 address.
+        struct in6_addr const *laddr6;
+
+        /// Remote IPv6 address.
+        struct in6_addr const *raddr6;
+
+        /// Local IP port.
+        in_port_t const *local_port;
+
+        /// Remote IP port.
+        in_port_t const *remote_port;
+
+        /// Network interface index.
+        int32_t *index;
+
+        /// MPTCP subflow backup priority status.
+        uint8_t *backup;
+
+        /**
+         * @brief MPTCP subflow related event error.
+         *
+         * @note This is value is equal to the @c sk_err member of
+         *       @c struct @c sock in the Linux kernel.
+         */
+        uint8_t *error;
+
+#ifdef MPTCPD_ENABLE_PM_NAME
```

Comment:
Is this a good time to purge this PM_NAME code?

Reply:
Sure.

Reply:
No strong preference. I suppose it's a separate topic which would make me lean toward a separate PR, but it's not worth lots of extra effort to split it out.